src/main/java/com/minecolonies/permissions/ColonyPermissionEventHandler.java

```diff
@@ -0,0 +1,240 @@
+            }
+        }
+        else
+        {
+            Log.getLogger().info("The opened entity is not inside `" + this.colony.getName() + "`");
+        }
+    }
+
+    /**
+     * ItemTossEvent handler.
+     *
+     * @param event ItemTossEvent
+     */
+    @SubscribeEvent
+    public void on(ItemTossEvent event)
+    {
+        final EntityPlayer playerIn = event.getPlayer();
+
+        Log.getLogger().debug(String.format("Check if playerIn `%s` is inside colony `%s`", playerIn.getName(), colony.getName()));
+        if (colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
+        {
+            Log.getLogger().debug(String.format("Check if playerIn `%s` has at least Rank `%s`", playerIn.getName(), colony.getName()));
+
+            Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
+
+            if (rank.ordinal() < Permissions.Rank.FRIEND.ordinal()) {
+                event.setResult(Event.Result.DENY);
+            }
+        } else
+        {
+            Log.getLogger().info(String.format("Player `%s` is not inside colony `%s`", playerIn.getName(), colony.getName()));
+        }
+    }
+
+    /**
+     * Template
+     */
+    public void on()
+    {
+        //
+    }
+
+    /**
+     * This method returns TRUE if this event should be denied.
+     *
+     * @param worldIn
+     * @param posIn
+     * @param playerIn
+     * @param blockState
+     * @return
+     */
+    private boolean checkBlockEventDenied(final World worldIn, final BlockPos posIn, final EntityPlayer playerIn, final IBlockState blockState)
+    {
+        Log.getLogger().debug("Check if coordinate is in colony `" + this.colony.getName() + "`");
+        if (colony.isCoordInColony(worldIn, posIn))
+        {
+            Log.getLogger().info("Coordinate is inside colony");
+
+            Log.getLogger().debug("Check if player `" + playerIn.getName() + "` is member of `" + this.colony.getName() + "`.");
+            if (colony.getPermissions().isColonyMember(playerIn)) {
+                Log.getLogger().info("Player `" + playerIn.getName() + "` is member");
+                final Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
+
+                Log.getLogger().debug("Check if player `" + playerIn.getName() + "` has at least rank `" + Permissions.Rank.NEUTRAL + "`");
+                if (rank.ordinal() >= Permissions.Rank.NEUTRAL.ordinal())
+                {
+                    Log.getLogger().info("Player `" + playerIn.getName() + "` has no permission to place/break a block in `" + this.colony.getName() + "`");
+
+                    return true;
+                }
+
+                Log.getLogger().debug("Check of block `" + blockState.getBlock().getRegistryName() + "` is instanceof `" + AbstractBlockHut.class.getSimpleName() + "`");
+                if (blockState.getBlock() instanceof AbstractBlockHut) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)